pageserver/src/repository.rs

```diff
@@ -176,6 +178,18 @@ pub trait Timeline {
 
     /// Get vector of prepared twophase transactions
     fn get_twophase(&self, lsn: Lsn) -> Result<Vec<TransactionId>>;
+
+    fn make_sender<'t>(&'t self) -> Box<dyn TimelineSender + 't>;
+}
+
+pub trait TimelineSender<'t> {
```

Comment:
This interface seems really low level. What is the content of a "message"?

I'm working on replacing the rocksdb implementation at https://github.com/zenithdb/zenith/pull/79, so I'm hoping that we don't introduce any new stuff that is tightly coupled with the current rocksdb implementation :-)